### PR TITLE
NCSDK-22231: Add support for new SUIT processor API

### PR DESCRIPTION
**Pull request by tomchy**

### CODE REVIEW

File: include/suit_processor.h [hunk 1]
```diff
@@ -206,4 +206,27 @@ struct suit_processor_state {
 	struct suit_seq_exec_state seq_stack[SUIT_MAX_SEQ_DEPTH];
 };
 
+
+/** @brief Poplate the manifest stack by loading a new envelope.
```
Comment (rafp83) [inline include/suit_processor.h:210]: Poplate => Populate

**Review by adsz-nordic — APPROVED** [review]
Found only typos - consider changing them if other changes are needed and if time permits. Otherwise it looks good.

**Review by rafp83 — APPROVED** [review]

File: include/suit_condition.h [hunk 2]
```diff
@@ -34,5 +34,14 @@ int suit_condition_component_slot(struct suit_processor_state *state,
 int suit_condition_abort(struct suit_processor_state *state,
 		struct suit_manifest_params *component_params);
 
+#ifndef SUIT_PLATFORM_LEGACY_API_SUPPORT
+/** Check a manifest component integrtity and signature. */
```
Comment (kszromek-nordic) [inline include/suit_condition.h:38]: Small typo in integrtity.

Comment (tomchy) [reply include/suit_condition.h:38]: I'll let the CI run, then force-push :slightly_smiling_face:

**Review by kszromek-nordic — APPROVED** [review]

**Pull request merged**